cmd/infra/azure/create.go

```diff
@@ -806,6 +806,8 @@ func assignServicePrincipalRoles(subscriptionID, managedResourceGroupName, nsgRe
 		scopes = append(scopes, vnetRG, dnsZoneRG)
 	case cpo:
 		scopes = append(scopes, nsgRG, vnetRG)
+	case azureFile:
```

Comment:
I don't believe this is needed. Could we remove this and retest? We are going to be stripping down these permissions soon.

Reply:
I tried with remove the permission previously, the file nfs volume provision got failed of permission issue.

Reply:
I'm mistaken. The operator on the HCP side will need permissions.